src/tools/dashboard/src/litani.rs

```diff
@@ -0,0 +1,96 @@
+pub struct Litani {
+    /// A buffer of the `spawn`ed Litani jobs so far. `Litani` takes some time
+    /// to execute each `add-job` command and executing thousands of them
+    /// sequentially takes a considerable amount of time. To speed up the
+    /// execution of those commands, we spawn those commands sequentially (as
+    /// normal). However, instead of `wait`ing for each process to terminate,
+    /// we add its handle to a buffer of the `spawn`ed processes and continue
+    /// with our program. Once we are done adding jobs, we wait for all of them
+    /// to terminate before we run the `run-build` command.
+    spawned_commands: Vec<Child>,
+}
+
+impl Litani {
+    /// Sets up a new [`Litani`] run.
+    pub fn init(project_name: &str, output_prefix: &Path, output_symlink: &Path) -> Self {
+        Command::new("litani")
+            .args([
+                "init",
+                "--project-name",
+                project_name,
+                "--output-prefix",
+                output_prefix.to_str().unwrap(),
+                "--output-symlink",
+                output_symlink.to_str().unwrap(),
+            ])
+            .spawn()
+            .unwrap()
+            .wait()
+            .unwrap();
+        Self { spawned_commands: Vec::new() }
+    }
+
+    /// Adds a single command with its dependencies.
+    pub fn add_job(
+        &mut self,
+        command: &Command,
+        description: &str,
+        pipeline: &str,
+        stage: &str,
+        exit_status: i32,
+    ) {
+        let mut job = Command::new("litani");
+        // The given command may contain additional env vars. Prepend those vars
+        // to the command before passing it to Litani.
+        let job_envs: HashMap<_, _> = job.get_envs().collect();
+        let mut new_envs = String::new();
+        command.get_envs().filter(|(k, _)| !job_envs.contains_key(k)).fold(
+            &mut new_envs,
+            |fmt, (k, v)| {
+                fmt.write_fmt(format_args!(
+                    "{}=\"{}\" ",
+                    k.to_str().unwrap(),
+                    v.unwrap().to_str().unwrap()
+                ))
+                .unwrap();
+                fmt
+            },
+        );
+        job.args([
+            "add-job",
+            "--command",
+            &format!("{}{:?}", new_envs, command),
+            "--description",
+            description,
+            "--pipeline-name",
+            pipeline,
+            "--ci-stage",
+            stage,
+            "--ok-returns",
+            &exit_status.to_string(),
+        ]);
+        // Start executing the command, but do not wait for it to terminate.
+        self.spawned_commands.push(job.spawn().unwrap());
+    }
+
+    /// Starts a [`Litani`] run.
+    pub fn run_build(&mut self) {
+        // Wait for all spawned processes to terminate.
+        for command in self.spawned_commands.iter_mut() {
+            command.wait().unwrap();
+        }
```

Comment:
What if they do not terminate? Are we using timeouts when invoking litani?

Reply:
We're not running the tests here. We're only adding them as jobs for Litani to run later. This command takes less around 0.1 second.  However, the `run-build` command after this may not terminate, but that's a good hint that we're missing a `--unwind` argument somewhere.

Reply:
Can we add a timeout to the run-build then? Litani should have that feature.

Reply:
Can do. Currently, individual jobs can take a `--timeout` flag which is often more useful, but added an [issue for this](https://github.com/awslabs/aws-build-accumulator/issues/34).